### ISSUE TITLE
How Sportsbet grew a React project from 2 devs to 20 (and beyond)

### ISSUE THREAD

**Issue by bent** [reactsydney/talks#79]
## Talk Date

July?

## Talk Title

How Sportsbet grew a React project from 2 devs to 20 (and beyond)

## Length

Can do anywhere from 20 to 30 minutes

## Twitter-sized Overview (140 characters)

The foundational tech we put in place up-front, what React/Redux concepts tripped-up and confused new devs, and how we dodged an SEO bullet.

## Detailed Overview

This talk will be the story of how Sportsbet structured a big React/React Native project so that lots of devs could work on it at once. This will include what things we put in-place up-front (vs those decisions that we could safely defer), what concepts in the React ecosystem commonly tripped-up and confused new devs (and how we dealt with it), and how one important detail that we missed nearly derailed the whole project.

I'll start with a quick overview of the project and it's motivations. Then I'll will move onto the foundation we had to lay up front for all of this to have a chance of achieving our ambitions, including things like type checking, testing strategies, CSS frameworks and important business logic. Next I'll cover the things that commonly confused developers new to React and Redux, and how we taught them the "React Way". Finally, I'll tell the hilarious/terrifying story of how we had to add SSR to our project relatively late, something I do not recommend.

In short, developers of all levels of experience will leave this talk with a sense of what it to consider when doing React 'in-the-large'.

**Comment by jesstelford** [comment]
Excellent proposal! It'll be fun to compare and contrast with my time at Domain, and I know there are plenty in the audience who will enjoy hearing your story.

Let's lock you in for July 2nd 🎉

**Comment by bent** [comment]
Sounds good, see you then!

**Comment by jesstelford** [comment]
Hey @bent July 2nd is rapidly approaching (this monday!) anything I can help with for prep?

**Comment by bent** [comment]
All good, as long as I have an hdmi cable I can plug into it’s all good.

I can do anywhere between 20 and 30 minutes. Do you have a preference?

**Comment by jesstelford** [comment]
Excellent, there's a whole array of plugs and adapters at the venue, so we'll be good for HDMI.

30 minutes + a bit of Q&A would work great.

**Comment by bent** [comment]
Sounds good, see you then!

Ben

**Comment by jesstelford** [comment]
Really awesome to see the community participation at the end of your talk - always a sign of something great!

Thanks for coming up, and for taking the time to give the talk before your flight :)

**Comment by bent** [comment]
No probs Jess, it was my pleasure. Next time I’ll get a later flight so I can actually hang around afterwards!
